bigtable/examples/bigtable_samples_instance_admin.cc

```diff
@@ -79,6 +79,36 @@ void CreateInstance(bigtable::InstanceAdmin instance_admin, int argc,
 }
 //! [create instance]
 
+//! [update instance]
+void UpdateInstance(bigtable::InstanceAdmin instance_admin, int argc,
+                    char* argv[]) {
+  if (argc != 2) {
+    throw Usage{"update-instance: <project-id> <instance-id>"};
+  }
+  std::string instance_id = ConsumeArg(argc, argv);
+  auto instance = instance_admin.GetInstance(instance_id);
+  // Modify the instance and prepare the mask with modified field
+  instance.set_display_name("Modified Display Name");
```

Comment:
I think the API here came out too low-level.  We should have a class like this:

```C++
class InstanceUpdates {
 public:
   void set_display_name(std::string x) {
     instance_.set_display_name(std::move(x));
     mask.add_paths("display_name"); // maybe check that it is not set already.
   }
  void set_foo(...) {
    instance_.set_foo(...);
   mask.add_paths("foo");
  }

 private:
   google::cloud::bigtable::v2::Instance updates_;
   google::protobuf::FieldMask mask;
};
```

Do you think it would be possible to write that?

Reply:
That sounds like a great idea.  Let's go with that.

Reply:
created `InstanceUpdateConfig` and used it in UpdateInstance